web3.eth.gas.pas

```diff
@@ -84,12 +111,78 @@ procedure getGasPrice(client: IWeb3; callback: TAsyncQuantity);
   end);
 end;
 
+procedure getBaseFeePerGas(client: IWeb3; callback: TAsyncQuantity);
+begin
+  web3.eth.getBlockByNumber(client, procedure(block: IBlock; err: IError)
+  begin
+    if Assigned(err) then
+      callback(0, err)
+    else
+      callback(block.baseFeePerGas, nil);
+  end);
+end;
+
+procedure getMaxPriorityFeePerGas(client: IWeb3; callback: TAsyncQuantity);
+begin
+  var adjustForSpeed := function(tip: BigInteger; speed: TGasPrice): BigInteger
+  begin
+    case speed of
+      Fastest: Result := TWei.Max(tip, 4);
+      Fast   : Result := TWei.Max(tip, 3);
+      Medium : Result := TWei.Max(tip, 2);
+      Low    : Result := 1;
```

Comment:
I would be hesitant to suggest "speeds".

You're probably best suited to offer a default, and allow overriding it. I think 2.5 is the new default most people use?

Reply:
@GregTheGreek We default to the `eth_maxPriorityFeePerGas` return value but in order to be backwards-compatible with existing dapps offering a low/medium/fast transaction speed, we're post-processing the `eth_maxPriorityFeePerGas` value here. I'm open to your suggestions.

Reply:
We maintain web3.js, so thats the opinion of the team. I don't think a library should do that